common/workunit/workunit.cpp

```diff
@@ -5416,6 +5416,34 @@ class CDaliWorkUnitFactory : public CWorkUnitFactory, implements IDaliClientShut
             StringArray unknownAttributes;
             Owned<CQueryOrFilter> orFilter;
 
+            void updateWUState(IPropertyTreeIterator* wuIter)
+            {
+                Owned<IRemoteConnection> conWorkUnitAborts = querySDS().connect("/WorkUnitAborts/", myProcessSession(), 0, SDS_LOCK_TIMEOUT);
+                IPropertyTree* WUAbortPTree = conWorkUnitAborts->queryRoot();
+                if (!WUAbortPTree)
+                    return;
+
+                ForEach(*wuIter)
+                {
+                    IPropertyTree& wu = wuIter->query();
+                    WUState state = (WUState) getEnum(&wu, "@state", states);
+                    switch (state)
+                    {
+                    case WUStateRunning:
+                    case WUStateDebugPaused:
+                    case WUStateDebugRunning:
+                    case WUStateBlocked:
+                    case WUStateCompiling:
+                        if (WUAbortPTree->queryPropTree(wu.queryName()))
+                            setEnum(&wu, "@state", WUStateAborting, states);
+                        break;
+                    case WUStateSubmitted:
+                        if (WUAbortPTree->queryPropTree(wu.queryName()))
+                            setEnum(&wu, "@state", WUStateAborted, states);
```

Comment:
as per above.

But also, in stead of re-coding the state logic here (the cases statement), calling IConstWorkunit::getState() on each workunit should in itself be enough (it checks state and aborting())